packages/forms/frontend/sirius-components-widget-reference/src/ReferencePropertySection.tsx

```diff
@@ -221,30 +222,57 @@ export const ReferencePropertySection = ({
     setModalDisplayed('browse');
   };
 
+  const onCreate = () => {
+    setModalDisplayed('create');
+  };
+
+  const addSelectedElements = (selectedElementIds: string[]): void => {
+    setModalDisplayed(null);
+    if (selectedElementIds) {
+      let newValueIds = [...selectedElementIds];
+
+      const variables = {
+        input: {
+          id: crypto.randomUUID(),
+          editingContextId,
+          representationId: formId,
+          referenceWidgetId: widget.id,
+          newValueIds,
+        },
+      };
+      editReference({ variables });
+    }
+  };
+
+  const addNewElement = (selectedElementId: string): void => {
+    setModalDisplayed(null);
+    if (selectedElementId) {
+      let newValueIds = widget.reference.manyValued
+        ? [...widget.referenceValues.map((value) => value.id), selectedElementId]
+        : [selectedElementId];
+
+      const variables = {
+        input: {
+          id: crypto.randomUUID(),
+          editingContextId,
+          representationId: formId,
+          referenceWidgetId: widget.id,
+          newValueIds,
+        },
+      };
+      editReference({ variables });
+    }
```

Comment:
Should the newly created element be selected?
I'm not sure, because: in the context of a Form representation it makes sense, so that the user can fill the properties of the newly created element in the Details view.
*But* in the context of the Details view itself that would _replace_ the details view the user was focused on.